dev-packages/request/src/node-request-service.ts

```diff
@@ -112,11 +119,17 @@ export class NodeRequestService implements RequestService {
                 } else {
                     const chunks: Uint8Array[] = [];
 
-                    res.on('data', chunk => {
+                    let stream: Stream = res;
+
+                    if (res.headers['content-encoding'] === 'gzip') {
+                        stream = stream.pipe(createGunzip());
+                    }
```

Comment:
I may be a bit too big a fan of ternaries, but here it seems reasonable, unless we anticipate other cases to handle?
```suggestion
                    const stream = res.headers['content-encoding'] === 'gzip' ? res.pipe(createGunzip()) : res;
```